src/sage/topology/simplicial_complex.py

```diff
@@ -1763,7 +1877,7 @@ def is_pseudomanifold(self):
         if d == 0:
             return len(self.facets()) == 2
         F = self.facets()
-        X = self.faces()[d-1]
+        X = self.faces().get(d-1) or []
```

Comment:
```suggestion
        X = self.faces()[d-1]
```
The `d==0` case was already handled separately, so you don't need to do any fancy checking here.